### DISCUSSION TITLE
A feedback about the recent change to hitting ESC at the end of a map

### DISCUSSION THREAD

**Discussion by rszyma** [ppy/osu#25257] [Q&A]
After changes in https://github.com/ppy/osu/pull/25097 hitting escape at the end of a map to exit feels a lot more "clunkier" than before the change. This is especially more noticable when "Hold-to-confirm activation time" is set to 0.

As to what I mean "clunkier": after I tap ESC it doesnt exit map immediately (like it was before) but is now delayed by like 1 sec. 

The solution I'd like to see is to exit a map immediately and if the result sceeen is not ready, just show a loading screen until it is.

**Comment by peppy** [comment]
Could you please show a video or something?

**Reply by rszyma** [reply]
Sure. The video is a bit over the GitHub limit, so I uploaded to discord: https://cdn.discordapp.com/attachments/891376262101233677/1167311083955028028/2023-10-27_05-43-41.mp4?ex=654daa2a&is=653b352a&hm=d393469ec63984f99d4bb885e277c8646daaad5b25406856e2fd91974a08680a&
Even though the vid turned out to be a bad quality (my computer got laggy), it's still clearly visible (and audible) when I press Esc and the end of the map.

**Reply by peppy** [reply]
Thanks. I don't suppose you still have your logs around from this session?

My hypothesis is that it's still waiting on the score submission process to complete, in which case your proposal of a loading screen is likely what we want to do (preferably on the results screen like stable does, rather than blocking the transition itself).

**Reply by rszyma** [reply]
> I don't suppose you still have your logs around from this session?

[database.log](https://github.com/ppy/osu/files/13192338/database.log)
[network.log](https://github.com/ppy/osu/files/13192339/network.log)
[performance.log](https://github.com/ppy/osu/files/13192340/performance.log)
[runtime.log](https://github.com/ppy/osu/files/13192341/runtime.log)

**Reply by KoleckOLP** [reply]
I have the same issue when I press esc at the end of the map it shows the hold to activate and only than it shows the result, l it's not instant as I expected. my hold to activate is also set to 0.

**Comment by rszyma** [comment]
Or maybe it's ignoring "Hold-to-confirm activation time" setting? Sometimes I see the Hold-to-confirm animation is playing and canceling when I'm tapping ESC button while it shouldn't play because I got this setting at 0 ms.

**Reply by peppy** [reply]
What you're seeing here is likely the result of pressing it multiple times, which will cause the animation to play but no action to be triggered. Likely not too relevant.

**Reply by Skubers** [reply]
Unburying this thread, as clicking the button multiple times It's not the case here, I can reproduce this easily on my side, I'm 100% sure I'm clicking Esc only one time, immediately when background goes brighter (due to last object being clicked).
I believe the reason that this issue is happening, is that the Back button (that is binded to Esc during Gameplay) still functions as the Hold-To-Pause action after clicking the last object, and stops being that action on Results screen. 
It should stop being a Hold-To-Pause button and become a Back button immediately after clicking the last object, right when the background goes bright.

(Uploaded video to hosting because github)
https://skubi.s-ul.eu/vP5TlOiJ
[compressed-logs.zip](https://github.com/ppy/osu/files/13717874/compressed-logs.zip)

**Comment by rszyma** [comment]
I'm sorry for bumping this, but I'd like to know if main devs are planning to pick this up anytime soon? It's been 4 months now since this regression landed.
